src/order/complete_lattice.lean

```diff
@@ -266,9 +266,17 @@ iff.intro
   (assume h a ha, top_unique $ h ▸ Inf_le ha)
   (assume h, top_unique $ le_Inf $ assume a ha, top_le_iff.2 $ h a ha)
 
+lemma Inf_eq_top_nonempty_eq_singleton {s : set α}
+  (h_inf : Inf s = ⊤) (hne : s.nonempty) : s = {⊤} :=
+by { rw set.eq_singleton_iff_nonempty_unique_mem, rw Inf_eq_top at h_inf, exact ⟨hne, h_inf⟩, }
```

Comment:
Golfs to
```suggestion
set.eq_singleton_iff_nonempty_unique_mem.mpr ⟨hne, Inf_eq_top.mp h_inf⟩
```
I'm sure you can do something similar with the one below

Reply:
I have mixed feelings about this one. I do use them but whenever I'm on the fence, I avoid term-mode proofs because they're harder to read (and, I claim, maintain).

Still, commit away if you like!